catalogs/catalog-fileset/src/test/java/org/apache/gravitino/catalog/fileset/TestFilesetCatalogOperations.java

```diff
@@ -1868,6 +1874,42 @@ public void testGetTargetLocation() throws IOException {
     }
   }
 
+  @Test
+  @Timeout(15)
+  void testGetFileSystemTimeoutThrowsException() throws Exception {
+    FieldUtils.writeField(
+        GravitinoEnv.getInstance(), "entityStore", new RelationalEntityStore(), true);
+
+    try (FilesetCatalogOperations filesetCatalogOperations = new FilesetCatalogOperations()) {
+      LocalFileSystemProvider localFileSystemProvider = Mockito.mock(LocalFileSystemProvider.class);
+      when(localFileSystemProvider.scheme()).thenReturn("file");
+      when(localFileSystemProvider.getFileSystem(Mockito.any(Path.class), Mockito.anyMap()))
+          .thenAnswer(
+              invocation -> {
+                // Block 100s, however, the timeout is set to 6s by default in
+                // FilesetCatalogOperations, so it's expected to be over within 10s
+                Awaitility.await().forever().until(() -> false);
+                return new LocalFileSystem();
+              });
+      Map<String, FileSystemProvider> fileSystemProviderMapOriginal = new HashMap<>();
+      fileSystemProviderMapOriginal.put("file", localFileSystemProvider);
+      FieldUtils.writeField(
+          filesetCatalogOperations, "fileSystemProvidersMap", fileSystemProviderMapOriginal, true);
+
+      FieldUtils.writeField(
+          filesetCatalogOperations, "propertiesMetadata", FILESET_PROPERTIES_METADATA, true);
+
+      // Test the following method should finish with 10s
+      long now = System.currentTimeMillis();
+      try {
+        filesetCatalogOperations.getFileSystem(new Path("file:///tmp"), ImmutableMap.of());
+      } catch (IOException e) {
+        long timeTake = System.currentTimeMillis() - now;
+        Assertions.assertTrue(timeTake <= 10000);
```

Comment:
I guess this will be very flaky if the test machine is under heavy load.

Reply:
I have used annotation `Timeout(15)` to replace it.